server/cluster/metadata/topology_manager.go

```diff
@@ -291,6 +307,64 @@ func (m *TopologyManagerImpl) RemoveTable(ctx context.Context, shardID storage.S
 	return nil
 }
 
+func (m *TopologyManagerImpl) GetTableShard(_ context.Context, table storage.Table) (storage.ShardID, bool) {
+	m.lock.RLock()
+	defer m.lock.RUnlock()
+
+	shardIDs, exists := m.tableShardMapping[table.ID]
+	if exists {
+		return shardIDs[0], true
+	}
+
+	return 0, false
+}
+
+func (m *TopologyManagerImpl) AssignTable(ctx context.Context, schemaID storage.SchemaID, tableName string, shardID storage.ShardID) error {
```

Comment:
This method actually assigns the table to a picked shard, so how about making the name more explicit:
```suggestion
func (m *TopologyManagerImpl) AssignTableToShard(ctx context.Context, schemaID storage.SchemaID, tableName string, shardID storage.ShardID) error {
```